#VRML_SIM R2025a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/robots/k-team/khepera1/protos/Khepera1.proto"

WorldInfo {
  info [
    "A Khepera robot equipped with a gripper"
  ]
  title "Khepera Gripper"
  CFM 0.001
  ERP 0.4
  basicTimeStep 10
  contactProperties [
    ContactProperties {
      material1 "gripper"
      material2 "stick"
      coulombFriction [
        10
      ]
    }
    ContactProperties {
      material1 "khepera"
      coulombFriction [
        0.1
      ]
    }
  ]
}
Viewpoint {
  orientation -0.2964333121842909 0.2945525594887188 0.9084966049062018 1.6729566941574279
  position 0.13967316686670384 -2.122450943839413 1.483237101739626
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  rotation 0 1 0 4.692820414042842e-06
  floorSize 1.5 1.5
}
DEF RED_STICK Solid {
  translation -0.368076 -0.373756 0.03
  children [
    Pose {
      rotation 1 0 0 3.14159
      children [
        DEF STICK_SHAPE Shape {
          appearance PBRAppearance {
            baseColor 1 0.155604 0.246125
            roughness 0.783569247
            metalness 0
          }
          geometry Cylinder {
            height 0.06
            radius 0.008
          }
        }
      ]
    }
  ]
  name "red stick"
  contactMaterial "stick"
  boundingObject Box {
    size 0.013 0.013 0.06
  }
  physics Physics {
    density 100
  }
}
DEF RED_STICK Solid {
  translation -0.123931 -0.360591 0.03
  children [
    Pose {
      rotation 1 0 0 3.14159
      children [
        DEF STICK_SHAPE Shape {
          appearance PBRAppearance {
            baseColor 1 0.155604 0.246125
            roughness 0.783569247
            metalness 0
          }
          geometry Cylinder {
            height 0.06
            radius 0.008
          }
        }
      ]
    }
  ]
  name "red stick(1)"
  contactMaterial "stick"
  boundingObject Box {
    size 0.013 0.013 0.06
  }
  physics Physics {
    density 100
  }
}
DEF RED_STICK Solid {
  translation 0.126936 -0.363571 0.03
  children [
    Pose {
      rotation 1 0 0 3.14159
      children [
        DEF STICK_SHAPE Shape {
          appearance PBRAppearance {
            baseColor 1 0.155604 0.246125
            roughness 0.783569247
            metalness 0
          }
          geometry Cylinder {
            height 0.06
            radius 0.008
          }
        }
      ]
    }
  ]
  name "red stick(2)"
  contactMaterial "stick"
  boundingObject Box {
    size 0.013 0.013 0.06
  }
  physics Physics {
    density 100
  }
}
DEF RED_STICK Solid {
  translation 0.370534 -0.361395 0.03
  children [
    Pose {
      rotation 1 0 0 3.14159
      children [
        DEF STICK_SHAPE Shape {
          appearance PBRAppearance {
            baseColor 1 0.155604 0.246125
            roughness 0.783569247
            metalness 0
          }
          geometry Cylinder {
            height 0.06
            radius 0.008
          }
        }
      ]
    }
  ]
  name "red stick(3)"
  contactMaterial "stick"
  boundingObject Box {
    size 0.013 0.013 0.06
  }
  physics Physics {
    density 100
  }
}
DEF BLUE_STICK Solid {
  translation -0.365309 0.35577 0.03
  children [
    Pose {
      rotation 1 0 0 3.14159
      children [
        DEF STICK_SHAPE Shape {
          appearance PBRAppearance {
            baseColor 0.0420081 0.0878462 0.932082
            roughness 0.783569247
            metalness 0
          }
          geometry Cylinder {
            height 0.06
            radius 0.008
          }
        }
      ]
    }
  ]
  name "blue stick"
  contactMaterial "stick"
  boundingObject Box {
    size 0.013 0.013 0.06
  }
  physics Physics {
    density 100
  }
}
DEF BLUE_STICK Solid {
  translation -0.120445 0.361784 0.03
  children [
    Pose {
      rotation 1 0 0 3.14159
      children [
        DEF STICK_SHAPE Shape {
          appearance PBRAppearance {
            baseColor 0.0420081 0.0878462 0.932082
            roughness 0.783569247
            metalness 0
          }
          geometry Cylinder {
            height 0.06
            radius 0.008
          }
        }
      ]
    }
  ]
  name "blue stick(1)"
  contactMaterial "stick"
  boundingObject Box {
    size 0.013 0.013 0.06
  }
  physics Physics {
    density 100
  }
}
DEF BLUE_STICK Solid {
  translation 0.130318 0.365399 0.03
  children [
    Pose {
      rotation 1 0 0 3.14159
      children [
        DEF STICK_SHAPE Shape {
          appearance PBRAppearance {
            baseColor 0.0420081 0.0878462 0.932082
            roughness 0.783569247
            metalness 0
          }
          geometry Cylinder {
            height 0.06
            radius 0.008
          }
        }
      ]
    }
  ]
  name "blue stick(2)"
  contactMaterial "stick"
  boundingObject Box {
    size 0.013 0.013 0.06
  }
  physics Physics {
    density 100
  }
}
DEF BLUE_STICK Solid {
  translation 0.381989 0.358112 0.03
  children [
    Pose {
      rotation 1 0 0 3.14159
      children [
        DEF STICK_SHAPE Shape {
          appearance PBRAppearance {
            baseColor 0.0420081 0.0878462 0.932082
            roughness 0.783569247
            metalness 0
          }
          geometry Cylinder {
            height 0.06
            radius 0.008
          }
        }
      ]
    }
  ]
  name "blue stick(3)"
  contactMaterial "stick"
  boundingObject Box {
    size 0.013 0.013 0.06
  }
  physics Physics {
    density 100
  }
}
Khepera1 {
  translation -0.380833 0.102898 2.03645e-14
  name "khepera"
  controller "my_controller"
  supervisor TRUE
  extensionSlot [
    DEF LIFT_HINGE_JOINT HingeJoint {
      jointParameters HingeJointParameters {
        position -1.4
        axis 0 1 0
        anchor -0.01 0 0.022
      }
      device [
        RotationalMotor {
          name "motor"
        }
      ]
      endPoint Solid {
        translation -0.01 0 0.022
        rotation 0.705814842414855 -0.7058148424148553 -0.06042198650976582 3.0208949999999994
        children [
          DEF RIGHT_GRIP SliderJoint {
            jointParameters JointParameters {
              axis 1 0 0
            }
            device [
              LinearMotor {
                name "right grip"
                maxVelocity 0.1
                maxPosition 0.029
                maxForce 100
              }
            ]
            endPoint Solid {
              translation 0 0 -0.048
              rotation 0 1 0 1.5708
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0.8 0.8 0.8
                    roughness 1
                    metalness 0
                  }
                  geometry IndexedFaceSet {
                    coord Coordinate {
                      point [
                        0 -0.004 0.004
                        0.03 -0.004 0.002
                        0.03 -0.004 0.0001
                        0 -0.004 0.0001
                        0 -0.004 0.004
                        0 0.004 0.004
                        0.03 0.004 0.002
                        0.03 0.004 0.0001
                        0 0.004 0.0001
                        0 0.004 0.004
                      ]
                    }
                    coordIndex [
                      4, 3, 2, 1, 0, -1, 5, 6, 7, 8, 9, -1, 0, 1, 6, 5, -1, 1, 2, 7, 6, -1, 2, 3, 8, 7, -1, 3, 4, 9, 8, -1
                    ]
                  }
                }
                DistanceSensor {
                  translation 0.015 0 0
                  rotation 0 1 0 1.5708
                  name "ds"
                  lookupTable [
                    0 0 0
                    0.05 1000 0
                  ]
                  type "infra-red"
                }
              ]
              name "right grip"
              contactMaterial "gripper"
              boundingObject Pose {
                translation 0.015 0 0.003
                children [
                  Box {
                    size 0.028 0.007 0.005
                  }
                ]
              }
              physics Physics {
                density 100
                centerOfMass [
                  0.01 0 0
                ]
              }
            }
          }
          DEF LEFT_GRIP SliderJoint {
            jointParameters JointParameters {
              axis -1 0 0
            }
            device [
              LinearMotor {
                name "left grip"
                maxVelocity 0.1
                maxPosition 0.029
                maxForce 100
              }
            ]
            endPoint Solid {
              translation 0 0 -0.048
              rotation 0 1 0 -1.5708
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 0.8 0.8 0.8
                    roughness 1
                    metalness 0
                  }
                  geometry IndexedFaceSet {
                    coord Coordinate {
                      point [
                        0 -0.004 0.0001
                        -0.03 -0.004 0.0001
                        -0.03 -0.004 0.002
                        0 -0.004 0.004
                        0 -0.004 0.0001
                        0 0.004 0.0001
                        -0.03 0.004 0.0001
                        -0.03 0.004 0.002
                        0 0.004 0.004
                        0 0.004 0.0001
                      ]
                    }
                    coordIndex [
                      4, 3, 2, 1, 0, -1, 5, 6, 7, 8, 9, -1, 0, 1, 6, 5, -1, 1, 2, 7, 6, -1, 2, 3, 8, 7, -1, 3, 4, 9, 8, -1
                    ]
                  }
                }
              ]
              name "left grip"
              contactMaterial "gripper"
              boundingObject Pose {
                translation -0.015 0 0.003
                children [
                  Box {
                    size 0.028 0.007 0.005
                  }
                ]
              }
              physics Physics {
                density 100
                centerOfMass [
                  -0.01 0 0
                ]
              }
            }
          }
          Shape {
            appearance PBRAppearance {
              baseColor 0.8 0.8 0.8
              roughness 1
              metalness 0
            }
            geometry IndexedFaceSet {
              coord Coordinate {
                point [
                  0.035 -0.004 0.006
                  0.035 -0.004 -0.048
                  -0.035 -0.004 -0.048
                  -0.035 -0.004 0.006
                  -0.025 -0.004 0.006
                  -0.025 -0.004 -0.03
                  -0.016 -0.004 -0.04
                  0.016 -0.004 -0.04
                  0.025 -0.004 -0.03
                  0.025 -0.004 0.006
                  0.035 -0.004 0.006
                  0.035 0.004 0.006
                  0.035 0.004 -0.048
                  -0.035 0.004 -0.048
                  -0.035 0.004 0.006
                  -0.025 0.004 0.006
                  -0.025 0.004 -0.03
                  -0.016 0.004 -0.04
                  0.016 0.004 -0.04
                  0.025 0.004 -0.03
                  0.025 0.004 0.006
                  0.035 0.004 0.006
                ]
              }
              coordIndex [
                10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, -1, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, -1, 0, 1, 12, 11, -1, 1, 2, 13, 12, -1, 2, 3, 14, 13, -1, 3, 4, 15, 14, -1, 4, 5, 16, 15, -1, 5, 6, 17, 16, -1, 6, 7, 18, 17, -1, 7, 8, 19, 18, -1, 8, 9, 20, 19, -1, 9, 10, 21, 20, -1
              ]
            }
          }
        ]
        boundingObject Pose {
          translation 0 0 -0.021
          children [
            Box {
              size 0.068 0.008 0.052
            }
          ]
        }
        physics Physics {
          density 100
          centerOfMass [
            0 0 -0.02
          ]
        }
      }
    }
    DEF GRIPPER_BASE Pose {
      translation -0.01 0 0.023
      rotation 0.5773502691896258 -0.5773502691896258 -0.5773502691896258 2.094395
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.8 0.8 0.8
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.05 0.01 0.01
          }
        }
      ]
    }
    Camera {
      translation -0.01 0 0.035
      rotation 1 0 0 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.70632 0.70632 0.70632
            roughness 1
            metalness 0
          }
          geometry DEF CAMERA Box {
            size 0.03 0.02 0.03
          }
        }
      ]
      boundingObject DEF CAMERA Box {
        size 0.03 0.02 0.03
      }
      physics Physics {
        density -1
        mass 0.02
      }
    }
  ]
}
